### PR TITLE
reimplement BPE tokenizer

### PR DESCRIPTION
**Pull request by pavel-esir**
CVS-136471

### CODE REVIEW

File: python/openvino_tokenizers/tokenizer_pipeline.py [hunk 1]
```diff
@@ -504,14 +514,27 @@ def finalize(self) -> None:
...
             fuse_unk=tokenizer_json["model"]["fuse_unk"] or False,
             suffix_indicator=tokenizer_json["model"]["continuing_subword_prefix"] or "",
             end_suffix=tokenizer_json["model"]["end_of_word_suffix"] or "",
             vocab=vocab,
             merges=tokenizer_json["model"]["merges"],
-            added_tokens={token["id"]: token["content"] for token in tokenizer_json["added_tokens"] if token["id"]},
+            added_tokens=added_tokens,
+            byte_fallback=tokenizer_json["model"]["byte_fallback"],
+            # cache_capacity=max(tokenizer_json["model"].get("cache_capacity", int(len(vocab) * VOCAB_SIZE_CACHE_PROPORTION)), MIN_CACHE_CAPACITY)
+            cache_capacity=20775
```
Comment (apaniukov) [inline python/openvino_tokenizers/tokenizer_pipeline.py:537]: Fix

Comment (pavel-esir) [reply python/openvino_tokenizers/tokenizer_pipeline.py:537]: Great catch. Thanks for noticing! I was suspecting i left this there

Comment (pavel-esir) [reply python/openvino_tokenizers/tokenizer_pipeline.py:537]: fixed

File: python/openvino_tokenizers/convert_tokenizer.py [hunk 2]
```diff
@@ -93,9 +94,13 @@ def convert_tokenizer(
         raise OVTypeError(f"Tokenizer type is not supported: {type(tokenizer_object)}")
 
     if isinstance(ov_tokenizers, tuple):
+        # import openvino as ov
+        # ov.runtime.save_model(ov_tokenizers[1], "openvino_detokenizer.xml")
+
```
Comment (apaniukov) [inline python/openvino_tokenizers/convert_tokenizer.py:99]: Fix

File: python/openvino_tokenizers/convert_tokenizer.py [hunk 3]
```diff
@@ -93,9 +94,13 @@ def convert_tokenizer(
...
         return (
             change_outputs_type(ov_tokenizers[0], tokenizer_output_type),
             change_inputs_type(ov_tokenizers[1], detokenizer_input_type),
         )
-
+    # import openvino as ov
+    # ov.runtime.save_model(ov_tokenizers, "openvino_tokenizer.xml")
```
Comment (apaniukov) [inline python/openvino_tokenizers/convert_tokenizer.py:105]: Fix

File: python/openvino_tokenizers/convert_tokenizer.py (hunk 2 above)
Comment (pavel-esir) [reply python/openvino_tokenizers/convert_tokenizer.py:99]: my bad 🤦in the very last moment suddenly pushed this. Will remove

File: python/openvino_tokenizers/convert_tokenizer.py (hunk 3 above)
Comment (pavel-esir) [reply python/openvino_tokenizers/convert_tokenizer.py]: fixed

File: python/openvino_tokenizers/convert_tokenizer.py [hunk 4]
```diff
@@ -41,6 +41,7 @@ def convert_tokenizer(
         )
 
         if isinstance(tokenizer_object, PreTrainedTokenizerBase):
+            # if False:
```
Comment (pavel-esir) [inline python/openvino_tokenizers/convert_tokenizer.py:44]: fix

Comment (pavel-esir) [reply python/openvino_tokenizers/convert_tokenizer.py]: done

**Review by apaniukov — APPROVED** [review]

**Pull request merged**